server/roomlogs.ts

```diff
@@ -231,16 +231,13 @@ export class Roomlog {
 		const roomlogPath = `logs/chat`;
 		const roomlogStreamExisted = this.roomlogStream !== null;
 		await this.destroy(false); // don't destroy modlog, since it's renamed later
-		await Promise.all([
+		const [roomlogExists, newRoomlogExists] = await Promise.all([
 			FS(roomlogPath + `/${this.roomid}`).exists(),
 			FS(roomlogPath + `/${newID}`).exists(),
-		]).then(([roomlogExists, newRoomlogExists]) => {
-			return Promise.all([
-				roomlogExists && !newRoomlogExists ?
-					FS(roomlogPath + `/${this.roomid}`).rename(roomlogPath + `/${newID}`) :
-					undefined,
-			]);
-		});
+		]);
+		if (roomlogExists && !newRoomlogExists) {
+			await FS(roomlogPath + `/${this.roomid}`).rename(roomlogPath + `/${newID}`);
+		}
```

Comment:
@AnnikaCodes I think most of these "you don't need to use `promise.then`" situations are you. This isn't a big deal and it's fine if you forget (unfortunately there's no way to make the linter detect these), but I hope this example helps illustrate how you would do this with `async`/`await`.

Reply:
Yeah, I'll try to convert to `await` logic in the future.